### ISSUE TITLE
Need your help!

### ISSUE THREAD

**Issue by orangeredbee** [maroxtn/SAINT-Transformer-riiid-kaggle#1]
Hi! 
Thank you for you code. However I have some trouble running your code.

As for data,  I download from ：
https://www.kaggle.com/c/riiid-test-answer-prediction/data?select=train.csv 
https://www.kaggle.com/c/riiid-test-answer-prediction/data?select=questions.csv
Then I run 'validation_split.py'  and  'preprocess.py' to preprocess the data.
After that I run the 'train.py'  to get result.

However ，with the config 
“n_epochs: 60,
seq_len: 100,
batch_size: 256, 
d_model: 64, 
decoder_layers: 2, 
encoder_layers: 2, 
dropout: 0.1, 
correct_start_token: 2, 
user_answer_start_token: 4” 
in your config.yaml,  the best result is just "Acc 0.694   ROC 0.740"

Then, I change the config as you told in "https://www.kaggle.com/abdessalemboukil/saint-training-inference-guide-39th-solution"  with 
"n_epochs: 60, 
seq_len: 100, 
batch_size: 512, 
d_model: 128, 
decoder_layers: 4, 
encoder_layers: 4, 
dropout: 0.1, 
correct_start_token: 2, 
user_answer_start_token: 4" .
The best result is just “ Acc 0.695   ROC 0.740”

It's too far form your reported result.  : (
Is there any wrong step in my experiment to run your code? I need your help!

**Comment by orangeredbee** [comment]
well， I just find where goes wrong. 
In "train.py“ :
train_loader = torch.utils.data.DataLoader(trainset, shuffle=True, batch_size=batch_size, drop_last=True)
val_loader = torch.utils.data.DataLoader(valset, shuffle=False, batch_size=batch_size, drop_last=False)

It missing  “collate_fn=collate_fn” as well as "class collate_fn"  you write in your Kaggle notebook. so that making a wrong dimension of all data. 

I hope you can fix it : )

**Comment by maroxtn** [comment]
Hey thanks reporting the mistake! Sorry I was away from my computer for the past few days.
Does it give the needed score after fixing it ?
